### DISCUSSION TITLE
How does airflow distributes schedulers on workers

### DISCUSSION THREAD

**Discussion by nitra-1** [apache/airflow#23801] [Q&A] [answered]
How does airflow balances the load on worker, if there are say 5 workers and 1 master.
How are the schedulers scheduled on the workers ?

**Comment by potiuk** [comment]
randomly. There is no load balancing. You can manually control the loads with celery queue mechanisms if you want to distribute the load.

**Reply by nitra-1** [reply]
can you help with, how to do it manually. As i understand,master does it on its own

**Comment by pingzh** [comment, marked as answer]
@nitra-1 are you talking about the CeleryExecutor?

**Reply by nitra-1** [reply]
Yes

**Reply by pingzh** [reply]
then yes, as in this https://github.com/apache/airflow/discussions/23801#discussioncomment-2783723,

for celery executor, airflow scheduler simply sends the tasks to the Celery with potentially different queues. If you want to prioritize some tasks over others, you can have different queues and assign different number of workers to different queues.

**Reply by nitra-1** [reply]
can we send some schedulers to a particular worker and only to that worker, and how ?

**Reply by pingzh** [reply]
i don't think there is a way with celeryexecutor. It looks like you want to label your worker. Could you please some context about why you want to do that?

**Reply by nitra-1** [reply]
I have project wise airflow implementation, i want to dedicate wroker for a particular project.
So if there are schedulers for project x , those will execute on worker x only.

**Reply by potiuk** [reply]
You can use Queues in Calery for that.

**Reply by potiuk** [reply]
https://airflow.apache.org/docs/apache-airflow/stable/executor/celery.html?highlight=queues#queues

**Reply by nitra-1** [reply]
Thank you!

**Reply by pingzh** [reply]
@nitra-1 if you think this has solved your question, could you please mark it answered? thanks
